src/SSVOpenHexagon/Core/MenuGame.cpp

```diff
@@ -1762,8 +1767,37 @@ void MenuGame::exitAction()
 
 void MenuGame::update(ssvu::FT mFT)
 {
-    steamManager.run_callbacks();
-    discordManager.run_callbacks();
+    // Update Steam Rich Presence
+    if(!hexagonGame.steamHung)
+    {
+        if(!steamManager.run_callbacks())
+        {
+            hexagonGame.steamAttempt += 1;
+            if(hexagonGame.steamAttempt > 20)
+            {
+                hexagonGame.steamHung = true;
+                ssvu::lo("Steam") << "Too many failed callbacks. Stopping "
+                                     "Rich Presence updates."
+                                  << "\n";
+            }
+        }
+    }
+
+    // Update Discord Rich Presence
+    if(!hexagonGame.discordHung)
+    {
+        if(!discordManager.run_callbacks())
+        {
+            hexagonGame.discordAttempt += 1;
+            if(hexagonGame.discordAttempt > 20)
+            {
+                hexagonGame.discordHung = true;
+                ssvu::lo("Discord") << "Too many failed callbacks. Stopping "
+                                       "Rich Presence updates."
+                                    << "\n";
+            }
+        }
+    }
```

Comment:
Code repetition with `HexagonGame`. This should be extracted to a separate function (or two) in a separate file, and then both `MenuGame` and `HexagonGame` can depend on it and invoke it.